internal/service/service.go

```diff
@@ -150,10 +150,16 @@ func (s *Service) decryptDriverData(key string, val interface{}) (ret interface{
 		return "ENC[" + parts[1] + "]", true
 	}
 
-	data := []byte(parts[1])
-	decoded, err := base64.StdEncoding.DecodeString(string(data))
-	if err != nil {
-		dipper.Logger.Panicf("encrypted data shoud be base64 encoded")
+	var decoded []byte
+
+	if parts[1][0] == '"' {
+		decoded = []byte(strings.Trim(parts[1], "\""))
```

Comment:
can this use `'"'` as well for consistency with the `if` condition (or vice-versa)?

This whole part seems a little obscure - maybe you can add a comment explaining what's happening here and what `parts[1][0]` is expected to be, or (better), make the whole thing less magic? Is this the case when we have a string starting with a leading double-quote instead of a base64 encrypted string?

Reply:
Right, if it starts with a double quote, then I don't use base64 to decode it.